### PR TITLE
144 major version in filenames

### PR DESCRIPTION
**Pull request by saschazelzer**
This is a PR for issue #144 with the following changes:

### Parallel major version installs

 - Executable now have the major version number appended, e.g. `usResourceCompiler3` and `usShell3`
 - Default include install dir is now: `${PREFIX}/CppMicroServices3`. E.g. `/usr/local/include/cppmicroservices3/cppmicroservices/Bundle.h`
 - On Windows, libraries have the major version number appended, e.g. `CppMicroServices3.dll` (on other platforms, names already contained the full version and symbolic links are set up)

### Parallel release/debug installs

 - By default, the debug postfix `d` is used, e.g. for debug builds on Linux we have `libCppMicroServicesd.so.3.0.0`, on Windows it is `CppMicroServices3d.dll`

### MinGW support

 - Fixed warnings and errors
 - Added Appveyor MinGW-w64 builds
 
### Other

 - Only build Release configurations on Appveyor to reduce work load
 - Library prefix (e.g. `lib`) and extensions (`.dll` etc.) are handled uniformly now via CMake configure_file of `GlobalConfig.h.in`.

### CODE REVIEW

File: framework/test/SharedLibraryTest.cpp [hunk 1]
```diff
@@ -35,40 +35,34 @@ int SharedLibraryTest(int /*argc*/, char* /*argv*/[])
   US_TEST_BEGIN("SharedLibraryTest");
 
 #ifdef US_PLATFORM_WINDOWS
-  const std::string LIB_SUFFIX = ".dll";
   const char PATH_SEPARATOR = '\\';
```
Comment (jeffdiclemente) [inline framework/test/SharedLibraryTest.cpp]: ```PATH_SEPARATOR``` could also be removed if ```DIR_SEP``` is used from ```TestingConfig.h```.

File: CMakeLists.txt [hunk 2]
```diff
@@ -361,8 +364,8 @@ endif()
 if(MINGW)
   # suppress warnings about auto imported symbols
   set(US_CXX_FLAGS "-Wl,--enable-auto-import ${US_CXX_FLAGS}")
-  # we need to define a Windows version
-  set(US_CXX_FLAGS "-D_WIN32_WINNT=0x0500 ${US_CXX_FLAGS}")
+  # we need to define a Windows version (Windows Vista)
+  set(US_CXX_FLAGS "-D_WIN32_WINNT=0x0600 -DNTDDI_VERSION=0x06000000 ${US_CXX_FLAGS}")
```
Comment (saschazelzer) [inline CMakeLists.txt]: I was using the lowest version that contained the necessary API to compile the project successfully (unfortunately, I do not remember which part required this). Officially, Windows 7 can still be our minimum supported version.

File: framework/test/SharedLibraryTest.cpp (hunk 1 above)
Comment (saschazelzer) [inline framework/test/SharedLibraryTest.cpp]: Good catch.

**Comment by saschazelzer** [comment]
Is anyone having some doubts / comments / opinions about this one?

**Comment by jeffdiclemente** [comment]
I'm currently reviewing. I will provide feedback/approval today

**Pull request merged**